src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const keyR of action.keysToRemove) {
+        delete copyState[keyR];
+      }
+    }
+
+    if (action.type === 'clear') {
+      Object.keys(copyState).forEach((key) => delete copyState[key]);
+    }
```

Comment:
[CODE STYLE]: Instead of multiple if statements, a `switch` statement should be used when dealing with a limited set of conditions. This will improve readability and organization of the code. Additionally, always include a `default` case in your `switch` statement for error handling.